test/parallel/test-domain-exit-dispose.js

```diff
@@ -26,7 +26,7 @@ function err() {
       console.error('This should not happen.');
       disposalFailed = true;
       process.exit(1);
-    });
+    }, 0);
```

Comment:
@qualquervalor2 Could you check if this works with `setImmediate()` instead of `setTimeout()`?

If it doesn't work, could you change the `0` to a `1`? The 0 becomes a 1 under the hood so it would be better to be explicit up front. :)